.github/workflows/test.yml

```diff
@@ -34,6 +34,13 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/setup-zeebe
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-integration-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-integration-
```

Comment:
:question: Are these minues at the end intended?

Reply:
yes, see the [sample](https://github.com/actions/cache/blob/main/examples.md#java---maven) and the docs on restore-keys https://github.com/actions/cache#usage

the behavior of the cache is that it would reuse other caches starting with that key on case it doesn't find an exact match for it's id.
So let's say you have a new branch and changed poms, so your key would be `linux-maven-integration-aladsasfdafda` however it's the first run, there is a cache miss for that exact key. However there is a cache from main where just the hash is different, it will reuse that one as it starts with `linux-maven-integration-` and afterwards create your own cache.